physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,915 @@
+   type(land_iau_state_type),          intent(inout) :: Land_IAU_state  
+   character(len=*),                   intent(  out) :: errmsg
+   integer,                            intent(  out) :: errflg
+
+   ! local
+   character(len=128)   :: fname
+   real(kind=kind_phys) :: sx, wx, wt, normfact, dtp
+   integer              :: k, nstep, kstep
+   integer              :: nfilesall, ntimesall
+   integer, allocatable :: idt(:) 
+   integer              :: nlon, nlat
+   logical              :: exists
+   integer              :: ncid, dimid, varid, status, IDIM
+   
+   real(kind=kind_phys) :: dt   !, rdt
+   integer              :: im, jm, km, nfiles, ntimes
+
+   integer :: is,  ie,  js,  je
+   integer :: npz
+   integer :: i, j  
+
+   !Errors messages handled through CCPP error handling variables
+   errmsg = ''
+   errflg = 0
+
+   npz = Land_IAU_Control%lsoil
+   km = Land_IAU_Control%lsoil
+   
+   is  = Land_IAU_Control%isc
+   ie  = is + Land_IAU_Control%nx-1
+   js  = Land_IAU_Control%jsc
+   je  = js + Land_IAU_Control%ny-1
+   nlon = Land_IAU_Control%nx
+   nlat = Land_IAU_Control%ny
+   !nblks = Land_IAU_Control%nblks
+   !blksz = Land_IAU_Control%blksz(1)
+
+   print*, "rank is ie js je nlon nlat", Land_IAU_Control%me, is, ie, js, je, nlon, nlat
+
+   ! allocate arrays that will hold iau state
+   allocate(Land_IAU_Data%stc_inc(nlon, nlat, km))
+   allocate(Land_IAU_Data%slc_inc(nlon, nlat, km))
+   ! allocate(Land_IAU_Data%snow_land_mask(nlon, nlat)) 
+   
+   ! allocate (Land_IAU_state%inc1%stc_inc(nlon, nlat, km))
+   ! allocate (Land_IAU_state%inc1%slc_inc(nlon, nlat, km))
+   ! allocate (Land_IAU_state%inc2%stc_inc(nlon, nlat, km))
+   ! allocate (Land_IAU_state%inc2%slc_inc(nlon, nlat, km)) 
+
+   Land_IAU_Data%hr1=Land_IAU_Control%iaufhrs(1)
+   Land_IAU_Data%wt = 1.0 ! IAU increment filter weights (default 1.0)
+   Land_IAU_Data%wt_normfact = 1.0
+   if (Land_IAU_Control%iau_filter_increments) then
+      ! compute increment filter weights, sum to obtain normalization factor
+      dtp=Land_IAU_Control%dtp
+      nstep = 0.5*Land_IAU_Control%iau_delthrs*3600/dtp
+      ! compute normalization factor for filter weights
+      normfact = 0.
+      do k=1,2*nstep+1
+         kstep = k-1-nstep
+         sx     = acos(-1.)*kstep/nstep
+         wx     = acos(-1.)*kstep/(nstep+1)
+         if (kstep .ne. 0) then
+            wt = sin(wx)/wx*sin(sx)/sx
+         else
+            wt = 1.0
+         endif
+         normfact = normfact + wt
+         if (Land_IAU_Control%me == Land_IAU_Control%mpi_root) print *,'filter wts',k,kstep,wt
+      enddo
+      Land_IAU_Data%wt_normfact = (2*nstep+1)/normfact
+   endif
+
+   ! increment files in fv3 tiles 
+   if (trim(Land_IAU_Control%iau_inc_files(1)) .eq. '' .or. Land_IAU_Control%iaufhrs(1) .lt. 0) then ! only 1 file expected
+      print*, "Error! in Land IAU init: increment file name is empty or iaufhrs(1) is negative"

```

Comment:
similar comment as previously here, also for consistency, probably should use either `print *,` or `write(6,*)` when writing similar lines, not both